### PR TITLE
Ruby: Missing dataflow for method defined under inner class

### PR DESCRIPTION
**Pull request by hiteshbedre**
Added a breaking test case, marked it as ignore: A method is defined under a inner class inheriting self, flow under that method is missing.

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/dataflow/DataFlowTests.scala [hunk 1]
```diff
@@ -2518,4 +2518,23 @@ class DataFlowTests extends RubyCode2CpgFixture(withPostProcessing = true, withD
     val sink   = cpg.call.name("puts").l
     sink.reachableByFlows(source).size shouldBe 2
   }
+
+  "dataflow in method defined under class << self block" ignore {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/dataflow/DataFlowTests.scala:2522]: Please add a comment explaining why it's being ignored (in this case, it's the missing translation for `class << self`) so it's clear when to enable this test.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 2]
```diff
@@ -1120,6 +1120,27 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
     callNode.columnNumber shouldBe Some(9)
   }
 
+  "method defined inside a class using << operator" in {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1123]: Wasn't this test case already added [here](https://github.com/joernio/joern/pull/3367/files#diff-45b026c78f0405d131600a5e173a4dc262cdaae9ad115dfae83deb1ac6f1a938R1123)?

**Review by xavierpinho — CHANGES_REQUESTED** [review]

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/dataflow/DataFlowTests.scala (hunk 1 above)
Comment (hiteshbedre) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/dataflow/DataFlowTests.scala:2522]: Added comment here: https://github.com/joernio/joern/pull/3368/commits/6abf690b5d8fd398096bf358f6a79727d3b48d2d

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala (hunk 2 above)
Comment (hiteshbedre) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala]: yes, don't know why commits are already part of master branch are still showing here, investigating.

**Comment by hiteshbedre** [comment]
Opened new PR: https://github.com/joernio/joern/pull/3396

**Pull request closed without merging**